corehq/apps/domain/models.py

```diff
@@ -688,6 +688,11 @@ def get_all(cls, include_docs=True):
     def get_all_names(cls):
         return sorted({d['key'] for d in cls.get_all(include_docs=False)})
 
+    @classmethod
+    def get_deleted_domain_names(cls):
+        domains = Domain.view("domain/deleted_domains", include_docs=False, reduce=False).all()
+        return sorted({d['key'] for d in domains})
```

Comment:
I don't think you want `sorted` here. That turns a set into a list, which is going to make the `in` check below much slower, and we don't actually need them in any specific order.

Reply:
Resolved in #31895 which I just merged into master/into this branch.